src/sql/src/plan.rs

```diff
@@ -781,6 +786,9 @@ pub struct ReadThenWritePlan {
 /// Generated by `ALTER ... IF EXISTS` if the named object did not exist.
 #[derive(Debug)]
 pub struct AlterNoopPlan {
+    /// Provide this name using the name AstInfo implementation, i.e.
+    /// `.to_ast_string()` or `to_string()` for `PartialName`.
+    pub name: String,
```

Comment:
I think it'll be cleaner in the long run to add a `add_notice` API to the `SessionCatalog`. Then the SQL planner can just attach whatever notices it likes as it plans, rather than needing to surface everything up to the adapter. It does mean will need a new `PlanNotice` type, and an `AdapterNotice::Plan(PlanNotice)` variant, like we do for errors.

The benefits are marginal for this change but I think will accrue quite a bit over time.